framework/framework.go

```diff
@@ -349,9 +351,13 @@ func ProcessDelete(ctx context.Context, obj interface{}, resources []Resource) e
 
 		var currentState interface{}
 		{
-			if canceledcontext.IsCanceled(ctx) {
+			if reconciliationcanceledcontext.IsCanceled(ctx) {
 				return nil
 			}
+			if resourcecanceledcontext.IsCanceled(ctx) {
+				ctx = resourcecanceledcontext.NewContext(ctx, make(chan struct{}))
+				continue
+			}
```

Comment:
When we cancel the complete loop we just return. When we cancel a single resource we want to continue with the next resource. We also have to make sure the context is initialized with a new channel we can close again. This is a tiny but important implementation detail. It is not really nice, but it works.